nodes/viz/metaball_out.py

```diff
@@ -0,0 +1,294 @@
+        row.operator(show_hide, text='', icon=icons('hide_render')).fn_name = 'hide_render'
+
+        col = layout.column(align=True)
+        col.prop(self, "meta_name", text='', icon='OUTLINER_OB_META')
+        col.operator(show_hide, text='Select Toggle').fn_name = 'mesh_select'
+
+        layout.prop_search(
+                self, 'material', bpy.data, 'materials',
+                text='',
+                icon='MATERIAL_DATA')
+        layout.prop(self, "threshold")
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_buttons(context, layout)
+        layout.prop(self, "view_resolution")
+        layout.prop(self, "render_resolution")
+
+    def copy(self, node):
+        self.n_id = ''
+        self.meta_name = self.get_next_name()
+        meta = self.create_metaball()
+        self.label = meta.name
+
+    def update_material(self):
+        if bpy.data.materials.get(self.material):
+            metaball_object = self.find_metaball()
+            if metaball_object:
+                metaball_object.active_material = bpy.data.materials[self.material]
+
+    def process(self):
+        if not self.activate:
+            return
+
+        if not self.inputs['Origins'].is_linked:
+            return
+
+        metaball_object = self.find_metaball()
+        if not metaball_object:
+            metaball_object = self.create_metaball()
+            print("Created new metaball")
+
+        self.update_material()
+
+        metaball_object.data.resolution = self.view_resolution
+        metaball_object.data.render_resolution = self.render_resolution
+        metaball_object.data.threshold = self.threshold
+
+        self.label = metaball_object.name
+
+        origins = self.inputs['Origins'].sv_get()
+        origins = Matrix_generate(origins)
+        radiuses = self.inputs['Radius'].sv_get()[0]
+        stiffnesses = self.inputs['Stiffness'].sv_get()[0]
+        negation = self.inputs['Negation'].sv_get(default=[[0]])[0]
+        types = self.inputs['Types'].sv_get()[0]
+
+        items = match_long_repeat([origins, radiuses, stiffnesses, negation, types])
+        items = list(zip(*items))
+
+        def setup_element(element, item):
+            (origin, radius, stiffness, negate, meta_type) = item
+            center, rotation, scale = origin.decompose()
+            element.co = center[:3]
+            if isinstance(meta_type, int):
+                if meta_type not in self.meta_type_by_id:
+                    raise Exception("`Types' input expects an integer number from 1 to 5")
+                meta_type = self.meta_type_by_id[meta_type]
+            element.type = meta_type
+            element.radius = radius
+            element.stiffness = stiffness
+            element.rotation = rotation
+            element.size_x = scale[0]
+            element.size_y = scale[1]
+            element.size_z = scale[2]
+            element.use_negative = bool(negate)
```

Comment:
bool() here not needed.

Reply:
You mean Python will translate it automatically? Well, probably yes. For me personally, it seems better if I do explicit conversion; but if there is some convention in Sverchok or in Python in general, i'll change it.

Reply:
python is such a circus:
True == 1
False == 0